taxi/forms.py

```diff
@@ -0,0 +1,51 @@
+from django import forms
+from django.contrib.admin.widgets import FilteredSelectMultiple
+from django.core.validators import RegexValidator
+from django.contrib.auth.forms import UserCreationForm
+
+from taxi.models import Driver, Car
+
+
+class DriverForm(UserCreationForm):
+    license_number = forms.CharField(
+        max_length=8,
+        validators=[
+            RegexValidator(
+                regex=r"^[A-Z]{3}\d{5}$",
+                message="For license number use 3 uppercase and 5 digits.",
+            )
+        ]
+    )
+
+    class Meta(UserCreationForm.Meta):
+        model = Driver
+        fields = UserCreationForm.Meta.fields + (
+            "first_name", "last_name", "license_number"
+        )
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    license_number = forms.CharField(
```

Comment:
you can play with inheritance here and avoid code repetition

Reply:
That's probably because you inherit DriverForm from UserCreationForm - use forms.ModelForm on parent class and set the fields explicitly like so : `("username",  "password","license_number",)`. Then you can inherit update form from DriverForm and reset Meta class only

Reply:
all the same this way, if I'm setting the "password" field to DriverForm, it appears in license update form as well: 
![image](https://github.com/user-attachments/assets/0ad18270-0ab2-4593-a377-39aa9968c826)
![image](https://github.com/user-attachments/assets/01dce67d-dd02-46ec-9116-2d2d19b0d032)

Reply:
OK. I have deleted unnecessary updating form, view and link. Now everything (even updating license) works like it should.